### PR TITLE
add a way to run several geocoder_tester

### PR DESCRIPTION
**Pull request by antoine-de**
migrate the bash script https://github.com/QwantResearch/geocoder-tester/blob/master/run_qwant.sh
to a python script.

this will hopefully makes it easier to have a clearer qwant's geocoder-tester fork

### CODE REVIEW

File: runner/tasks.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            res.append(r)
+
+    report = "\n".join(_pretty_print(res, REPORT_COLUMN))
+    logging.info(report)
+
+    report_file = os.path.join(ctx.output_dir, f"report.log")
+    with open(report_file, "w") as log_file:
+        log_file.writelines(f"report on '{name}'\n")
+        log_file.writelines(f"queries make on {url} | version = {version} \n")
+        log_file.writelines(report)
```
Comment (amatissart) [inline runner/tasks.py]: you probably want to use `write` instead of `writelines` (which expects an iterable)

File: runner/Pipfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+
+[packages]
+
+invoke = "*"
+pytest = "*"
+pytest-xdist = "*"
+requests = "*"
+pyaml = "*"
+geopy = "*"
```
Comment (amatissart) [inline runner/Pipfile:17]: why are `pyaml` and `geopy` needed ?

File: runner/tasks.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+RESULT_PATTERN = re.compile(
+    "===+( (?P<failed>\d+) failed,)? (?P<success>\d+) passed.*in (?P<time>.*) seconds.*"
+)
+
+
+def _safe_cast(val, to_type):
+    try:
+        return to_type(val)
+    except (ValueError, TypeError):
+        return to_type()
```
Comment (amatissart) [inline runner/tasks.py]: Is it really what we want ?  
For example an invalid string that should contain an int will be silently parsed as 0 :thinking:

File: runner/Pipfile (hunk 2 above)
Comment (antoine-de) [reply runner/Pipfile:17]: it's the geocoder dependencies that we load :confused:

File: runner/tasks.py (hunk 3 above)
Comment (antoine-de) [reply runner/tasks.py]: yes since it's used only to parse the pytest results, It's useful to get the pretty print

File: runner/tasks.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        # extra chars).
+        tpl = "|".join(cell.format(key=key, length=l[key]) for key in keys)
+        out.append(tpl.format(**row))
+    return out
+
+
+RESULT_LINE_PATTERN = re.compile("===+.*===+")
+
+RESULT_PATTERN = re.compile(
+    "===+( (?P<failed>\d+) failed,)? ((?P<success>\d+) passed)?.*in (?P<time>.*) seconds.*"
```
Comment (amatissart) [inline runner/tasks.py]: Very minor note: this still does not match the number of failures in
`========================= 952 failed in 17.06 seconds ==========================`
(the comma should be optionnal)

**Review by amatissart — APPROVED** [review]

**Pull request merged**